src/java.base/share/classes/java/util/concurrent/ThreadLocalRandom.java

```diff
@@ -1066,22 +422,55 @@ private Object readResolve() {
     private static final long INHERITEDACCESSCONTROLCONTEXT
         = U.objectFieldOffset(Thread.class, "inheritedAccessControlContext");
 
-    /** Rarely-used holder for the second of a pair of Gaussians */
-    private static final ThreadLocal<Double> nextLocalGaussian =
-        new ThreadLocal<>();
-
     /** Generates per-thread initialization/probe field */
     private static final AtomicInteger probeGenerator = new AtomicInteger();
 
     /** The common ThreadLocalRandom */
     static final ThreadLocalRandom instance = new ThreadLocalRandom();
 
+    private static final class ThreadLocalRandomProxy extends Random {
+        @java.io.Serial
+        static final long serialVersionUID = 0L;
+
+       public int nextInt() {
+           return ThreadLocalRandom.current().nextInt();
+       }
+
+       public long nextLong() {
+           return ThreadLocalRandom.current().nextLong();
+       }
+    }
+
+    private static final AbstractSpliteratorGenerator proxy = new ThreadLocalRandomProxy();
```

Comment:
should be declared inside ThreadLocalRandomProxy, so the value is only initialized when proxy() is called

Reply:
agree